### PR TITLE
[NETBEANS-54] Module Review libs.freemarker

### PR DESCRIPTION
**Pull request by vieiro**
- Added maven coordinates to external/binaries_list for freemarker 2.3.19 (BSD 3?)
  - Added headers to two source files (FreemarkerEngine.java & FreemarkerFactory.java) and to one test file.
  - All unit tests passing.
  - Not adding license headers to the following test files (as some tests would fail):
    -  ./test/unit/data/golden/GeneratedMethodBody.java
    -  ./test/unit/data/golden/ClassWithoutReplacements.java
    -  ./test/unit/data/golden/GeneratedMethodBody2.java
    -  ./test/unit/data/golden/ForceNoReplacements.java
    -  ./test/unit/data/golden/SimpleReplacements.java
    -  ./test/unit/data/licenseheader.txt
    -  ./test/unit/data/templates/GeneratedMethodBody.java
    -  ./test/unit/data/templates/ClassWithoutReplacements.java
    -  ./test/unit/data/templates/SimpleReplacements.java

### CODE REVIEW

File: libs.freemarker/external/binaries-list [hunk 1]
```diff
@@ -14,4 +14,4 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-5EEAAC41164FEBCB79C73BEBD678A7B6C10C3E80 freemarker-2.3.19.jar
+A251045E5FADD02824D17F1AA8C412ACCF1AA1C9 org.freemarker:freemarker:2.3.19
```
Comment (matthiasblaesing) [inline libs.freemarker/external/binaries-list]: This is in the license:

> 2.  The end-user documentation included with the redistribution, if any, must
>     include the following acknowlegement:
>       "This product includes software developed by the
>       Visigoth Software Society (http://www.visigoths.org/)."
>     Alternately, this acknowlegement may appear in the software itself, if and
> wherever such third-party acknowlegements normally appear.

I would this include a "-notice" file, that in the end should end up in the apache NOTICE file. From my reading this is similar to the apache NOTICE file, that needs to be redistributed.

File: libs.freemarker/external/freemarker-2.3.19-notice.txt [hunk 2]
```diff
@@ -1,6 +1,35 @@
-Apache FreeMarker (incubating)
-Copyright 2015-2017 The Apache Software Foundation
+Copyright (c) 2003 The Visigoth Software Society. All rights reserved.
 
-This product includes software developed at
-The Apache Software Foundation (http://www.apache.org/).
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
```
Comment (matthiasblaesing) [inline libs.freemarker/external/freemarker-2.3.19-notice.txt]: I would not go for the full license in the notice. I would just add the required part (see section 2 of the license):

> This product includes software developed by the
> Visigoth Software Society (http://www.visigoths.org/).

For updates please also squash the changes together - there will now be at least four changeset for more or less one change.

**Comment by matthiasblaesing** [comment]
Thank you - I squashed your changes together and added an exclusion in the rat report for the test-data.

**Comment by vieiro** [comment]
Thank you, I'll try to add those test files to the exclusion list myself in future PRs.

**Pull request closed without merging**